command/exec.go

```diff
@@ -0,0 +1,225 @@
+package command
+
+import (
+	"context"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/evergreen/subprocess"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/google/shlex"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+type simpleExec struct {
+	CommandName string            `mapstructure:"command_name"`
+	Args        []string          `mapstructure:"args"`
+	Env         map[string]string `mapstructure:"env"`
+
+	Command string `mapstructure:"command"`
+
+	// Background, if set to true, prevents shell code/output from
+	// waiting for the script to complete and immediately returns
+	// to the caller
+	Background bool `mapstructure:"background"`
+
+	// Silent, if set to true, prevents shell code/output from being
+	// logged to the agent's task logs. This can be used to avoid
+	// exposing sensitive expansion parameters and keys.
+	Silent bool `mapstructure:"silent"`
+
+	// SystemLog if set will write the shell command's output to the system logs, instead of the
+	// task logs. This can be used to collect diagnostic data in the background of a running task.
+	SystemLog bool `mapstructure:"system_log"`
+
+	// WorkingDir is the working directory to start the shell in.
+	WorkingDir string `mapstructure:"working_dir"`
+
+	// IgnoreStandardOutput and IgnoreStandardError allow users to
+	// elect to ignore either standard out and/or standard output.
+	IgnoreStandardOutput bool `mapstructure:"ignore_standard_out"`
+	IgnoreStandardError  bool `mapstructure:"ignore_standard_error"`
+
+	// RedirectStandardErrorToOutput allows you to capture
+	// standard error in the same stream as standard output. This
+	// improves the synchronization of these streams.
+	RedirectStandardErrorToOutput bool `mapstructure:"redirect_standard_error_to_output"`
+
+	// ContinueOnError determines whether or not a failed return code
+	// should cause the task to be marked as failed. Setting this to true
+	// allows following commands to execute even if this shell command fails.
+	ContinueOnError bool `mapstructure:"continue_on_err"`
+
+	base
+}
+
+func simpleExecFactory() Command   { return &simpleExec{} }
+func (c *simpleExec) Name() string { return "simple.exec" }
+
+func (c *simpleExec) ParseParams(params map[string]interface{}) error {
+	err := mapstructure.Decode(params, c)
+	if err != nil {
+		return errors.Wrapf(err, "error decoding %v params", c.Name())
+	}
+
+	if c.Command != "" {
+		if c.CommandName != "" || len(c.Args) > 0 {
```

Comment:
Should `.c.CommandName` actually be `c.Command`? (based on the fact that that's what you're splitting on 77). I'm actually not sure what the difference between `command` and `command_name` is.

Reply:
"command" is supposed to be a string form that we're will decompose into a binary name and a slice of args.